### ISSUE TITLE
Acroform and/or Signature Object

### ISSUE THREAD

**Issue by Kyoku57** [24eme/signaturepdf#28]
Hi, the result PDF doesn't seem to include a Signature Field / Acroform associated to the rendered signature.

Do you plan to implement this kind of functionality ? Perhaps with the inclusion of calls to a backend based on https://pyhanko.readthedocs.io/en/latest/lib-guide/sig-fields.html 

In any case, good job ;-)

**Comment by teymour** [comment]
Good idea ! If you can share a PDF with this kind of feature, it whould help :)

**Comment by Kyoku57** [comment]
Hi, 

You'll find 3 versions of the PDF.
- without signature field (original): [document.pdf](https://github.com/24eme/signaturepdf/files/8487479/document.pdf)
- with signature field (Sig1): [document_prepared_with_signatureField.pdf](https://github.com/24eme/signaturepdf/files/8487478/document_prepared_with_signatureField.pdf)
- with the user signature: [document_signed.pdf](https://github.com/24eme/signaturepdf/files/8487477/document_signed.pdf)

The script in Quick'n'Dirty mode : 
- copy the document.pdf to document_prepared_with_signatureField.pdf
- launch the python script (needs pip install pyhanko)

```python
from pyhanko.sign.fields import SigFieldSpec, append_signature_field
from pyhanko.pdf_utils.incremental_writer import IncrementalPdfFileWriter

with open('document_prepared_with_signatureField.pdf', 'rb+') as doc:
    w = IncrementalPdfFileWriter(doc)
    signature_field = SigFieldSpec(sig_field_name="Sig1", on_page=0, box=(50,50,300,80))
    append_signature_field(w, signature_field)
    w.write_in_place()
```
Your file is updated with a empty Signature Field named Sig1
![image](https://user-images.githubusercontent.com/640683/163344350-6464e9cb-9e3f-413d-80ad-fb252fef0aa6.png)

In Acrobat, if you click on it, you can sign with a qualified device, or with a "Visual Signature" qualified by Adobe (no legal value but it's works here). With an API, you can sign the signature field (with a linked device, or a keystore .. that contains the private key and certificate needed to sign ;-))

For this kind of stuff, you need to use a private key somewhere to "sign" you signature Field, so ... you need to realize this signature with a backend service or a YubiKey/Browser capability. 

I'm not sure is the principale aim of your cool tool, but if you are interesting in go deeper with electronic signature .... 

![image](https://user-images.githubusercontent.com/640683/163345623-2c890d33-b645-45f3-aeb7-f2517e5594ea.png)

:D

**Comment by teymour** [comment]
Thanks you !

The signature field is detected as a data fields by pdftk : 

    $ pdftk document_prepared_with_signatureField.pdf dump_data_fields output /dev/stdout 
    ---
    FieldType: Signature
    FieldName: Sig1
    FieldFlags: 0
    FieldJustification: Left

With data fields pdf, the action menu could be transform in field menu : the user could enter the different fields of the pdf.
